py/desitarget/mock/build.py

```diff
@@ -818,15 +826,20 @@ def _update_spectra_status(result):
     except:
         nsky = 0
 
+    #compute target healpixel number
+    targpix = radec2pix(nside, targets['RA'], targets['DEC'])
     targetid = encode_targetid(objid=targets['BRICK_OBJID'],
-                               brickid=targets['BRICKID'], mock=1)
+                               brickid=targpix, mock=1)
```

Comment:
This is OK for now but in the future we will want to keep `BRICKID` to what it gets from the `brick_info` dictionary and ensure earlier in the code that `OBJID` is unique, thereby ensuring that `TARGETID` is always unique.  

We should also really refactor this piece of the code out and write a unit test to ensure `TARGETID` is never duplicated.